src/ProjectTemplates/Web.ProjectTemplates/content/WebApi-CSharp/Controllers/WeatherController.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+#if (!NoAuth)
+using Microsoft.AspNetCore.Authorization;
+#endif
+using Microsoft.AspNetCore.Mvc;
+
+namespace Company.WebApplication1.Controllers
+{
+#if (!NoAuth)
+    [Authorize]
+#endif
+
+    [Route("api/SampleData/[controller]")]
+    [ApiController]
+    public class WeatherController : ControllerBase
+    {
+        [HttpGet]
+        public ActionResult<WeatherResult> GetWeatherForecasts(string location, TemperatureUnit unit)
+        {
+            var rng = new Random();
```

Comment:
I was about to say that generally you [want to re-use](https://docs.microsoft.com/en-us/dotnet/api/system.random?redirectedfrom=MSDN&view=netcore-3.0#avoiding-multiple-instantiations) the same `Random` instance rather than re-creating it on each call, but it turns out that `Random` [is not thread-safe](https://docs.microsoft.com/en-us/dotnet/api/system.random?redirectedfrom=MSDN&view=netcore-3.0#the-systemrandom-class-and-thread-safety), meaning that you wouldn't want to use just the one instance without a doing your own locking scheme.

All of that's just background for anyone interested though, there's no need to change what you have here since it's just a sample and doesn't need to be production ready.